de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java

```diff
@@ -160,4 +164,47 @@ protected void addEditingDomainAndRepository() {
 	protected String getProjectName() {
 		return TEST_PROJECT_NAME + "_" + this.getClass().getSimpleName();
 	}
+	
+	
+	/**
+	 * Simple method to execute code as part of a recording command
+	 * @param runnable the lambda to be executed with no parameters and no return value. 
+	 */
+	protected <T> void executeAsCommand(Runnable runnable) {
+		// Now wrap the lambda into a recording command and execute it
+		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
+			@Override
+			protected void doExecute() {
+				runnable.run();
+			}
+		});
+	}
+		
+	/**
+	 * Simple method to execute code as part of a recording command
+	 * @param supplier the lambda to be executed with no parameters. 
+	 * @return the result of the executed lambda
+	 */
+	protected <T> T executeAsCommand(Supplier<T> supplier) {
+		
+		// Simple class to store object of the given generic type T
+		class Store {

```

Comment:
I guess this Store class is introduced to avoid the effectivly final requirement since you want to save the result from the doExecute call?
You may disagree, but I would prefer using `final List<T> objects = new ArrayList<>();` and then `objects.add(supplier.get())` simple since it avoids introducing some new magic class.

Reply:
Decided to go with a vector